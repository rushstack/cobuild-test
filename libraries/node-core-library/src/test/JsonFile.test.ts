// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import { JsonFile } from '../JsonFile';

// The PosixModeBits are intended to be used with bitwise operations.
/* eslint-disable no-bitwise */

describe(JsonFile.name, () => {
  it('adds a header comment', () => {
    expect(
      JsonFile.stringify(
        { abc: 123 },
        {
          headerComment: '// header\n// comment'
        }
      )
    ).toMatchSnapshot();
  });
  it('adds an empty header comment', () => {
    expect(
      JsonFile.stringify(
        { abc: 123 },
        {
          headerComment: ''
        }
      )
    ).toMatchSnapshot();
  });
  it('allows undefined values when asked', () => {
    expect(
      JsonFile.stringify(
        { abc: undefined },
        {
          ignoreUndefinedValues: true
        }
      )
    ).toMatchSnapshot();

    expect(
      JsonFile.stringify(
        { abc: undefined },
        {
          ignoreUndefinedValues: true,
          prettyFormatting: true
        }
      )
    ).toMatchSnapshot();
  });
});
