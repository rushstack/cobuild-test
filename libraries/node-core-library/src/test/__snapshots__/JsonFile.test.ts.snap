// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonFile adds a header comment 1`] = `
"// header
// comment
{
  \\"abc\\": 123
}
"
`;

exports[`JsonFile adds an empty header comment 1`] = `
"{
  \\"abc\\": 123
}
"
`;

exports[`JsonFile allows undefined values when asked 1`] = `
"{}
"
`;

exports[`JsonFile allows undefined values when asked 2`] = `
"{}
"
`;
